File: kotlin/lesson 3/TwoSum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = mutableMapOf<Int, Int>()

        nums.forEachIndexed { i, element -> 
            val diff = target - element
            if (map.containsKey(diff)) return intArrayOf(map.get(diff)!!, i)
            map.put(element, i)
        }

        return intArrayOf()
    }
}

File: javascript/lesson 3/twoSum.js
function twoSum(nums, target) {
    const map = new Map();
    for (let i = 0; i < nums.length; i++) {
        const diff = target - nums[i];
        if (map.get(diff) != null) return [map.get(diff), i];
        map.set(nums[i], i);
    }
}
